src/script/sign.h

```diff
@@ -67,9 +67,238 @@ struct SignatureData {
     explicit SignatureData(const CScript& script) : scriptSig(script) {}
 };
 
+/** A version of CTransaction with the PSBT format*/
+struct PartiallySignedTransaction
+{
+    CMutableTransaction tx;
+    std::map<uint160, CScript> redeem_scripts;
+    std::map<uint256, CScript> witness_scripts;
+    std::vector<std::map<CPubKey, std::vector<unsigned char>>> partial_sigs;
+    std::map<COutPoint, CTransaction> non_witness_utxos;
+    std::map<COutPoint, CTxOut> witness_utxos;
```

Comment:
I think you'll need a map to represent all the key/values that the deserializer doesn't understand (which just get serialized out again).

For the per-input things, perhaps have another struct (`PartiallySignedInput`?), and have a map from ints (positions) to those, instead of identifying those by COutPoint? I think that will simplify serialize/deserialize.